test/Index/roles-contained.swift

```diff
@@ -0,0 +1,142 @@
+// RUN: %target-swift-ide-test -print-indexed-symbols -include-locals -source-filename %s | %FileCheck %s
```

Comment:
Some more test ideas:
  - TuplePattern with an init that returns a tuple, ie. `let (a, b): (Int, String) = returnsTuple()`
  - Multiple patterns/inits in PDB, ie. `let (a, b) = foo(), (c, d) = bar()`